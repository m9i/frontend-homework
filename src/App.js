import "./stylesheets/index.scss";

import DropdownLink from "./components/DropdownLink/DropdownLink";
import DropdownItem from "./components/DropdownItem/DropdownItem";

const App = () => (
  <div>
    <nav className="nav">
      <DropdownLink />
      <DropdownItem />
    </nav>
  </div>
);

export default App;
