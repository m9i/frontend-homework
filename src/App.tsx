import './stylesheets/index.scss'

import DropdownLink from './DropdownLink'

const App = () => (
  <nav className="nav">
    <DropdownLink />
  </nav>
)

export default App
