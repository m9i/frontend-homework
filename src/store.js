import { createStore, applyMiddleware } from "redux";
import thunkMiddleware from "redux-thunk";
import { composeWithDevTools } from "redux-devtools-extension";
import reducer from "./reducers/reducer";

const composedEnhancer = composeWithDevTools(applyMiddleware(thunkMiddleware));
const store = createStore(reducer, composedEnhancer);

store.subscribe(() => {
  console.log("subscribe", store.getState());
});
export default store;
