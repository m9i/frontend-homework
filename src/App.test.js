import React from "react"
import App from "./App"
import { shallow } from "enzyme"
import { Provider } from "react-redux"
import store from "./store"

describe("rendering components", () => {
  it("Renders App component without crashing", () => {
    const AppComponent = shallow(
      <Provider store={store}>
        <App />
      </Provider>
    ).dive()
    expect(AppComponent)
  })

  it("snapshot", () => {
    const AppComponent = shallow(
      <Provider store={store}>
        <App />
      </Provider>
    ).dive()
    expect(AppComponent).toMatchSnapshot()
  })
})

describe("App", () => {})
