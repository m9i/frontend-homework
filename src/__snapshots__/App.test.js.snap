// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot 1`] = `<Connect(App) />`;

exports[`rendering components snapshot 1`] = `<Connect(App) />`;
